dbms/src/Functions/FunctionsHashing.h

```diff
@@ -956,6 +956,136 @@ class FunctionURLHash : public IFunction
     }
 };
 
+struct JavaHashImpl
+{
+    static Int32 apply(const char * data, const size_t size)
+    {
+        Int32 h = 0;
+        for (int i = 0; i < (int)size; ++i)
+        {
+            h = 31 * h + data[i];
```

Comment:
Using `data[i]` is incorrect, because signed-ness of `char` may be different on different platforms (x86, ARM).

Reply:
how to fix this problem? can we just simply transform data[i] to (int)data[i]?